dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/BitBucketSourceCodeRepo.java

```diff
@@ -223,19 +223,14 @@ private <T> T getArbitraryURL(String url, GenericType<T> type) throws ApiExcepti
     public SourceFile getSourceFile(String path, String repositoryId, String branch, DescriptorLanguage.FileType type) {
         // TODO: should we even be creating a sourcefile before checking that it is valid?
         // I think it is fine since in the next part we just check that source file has content or not (no content is like null)
-        SourceFile file = null;
         String content = this.readFile(repositoryId, path, branch);
 
-        if (!Strings.isNullOrEmpty(content)) {
-            file = new SourceFile();
-            // Grab content from found file
-            // do not censor invalid versions to match github expected behaviour
-            file.setType(type);
-            file.setContent(content);
-            file.setPath(path);
-            file.setAbsolutePath(path);
+        if (Strings.isNullOrEmpty(content)) {
+            return null;
         }
-        return file;
+        // Grab content from found file
+        // do not censor invalid versions to match github expected behaviour
+        return SourceFileHelper.create(type, content, path, path);
```

Comment:
Oh and I do agree this, or its builder replacement should be in the `SourceFile` class